core/services/keeper/upkeep_executer.go

```diff
@@ -196,13 +190,21 @@ func (ex *UpkeepExecuter) execute(upkeep UpkeepRegistration, headNumber int64, d
 			"performUpkeepGasLimit": upkeep.ExecuteGas + ex.orm.config.KeeperRegistryPerformGasOverhead(),
 			"checkUpkeepGasLimit": ex.config.KeeperRegistryCheckGasOverhead() + uint64(upkeep.Registry.CheckGas) +
 				ex.config.KeeperRegistryPerformGasOverhead() + upkeep.ExecuteGas,
-			"gasPrice":   gasPrice,
-			"gasTipCap":  fee.TipCap,
-			"gasFeeCap":  fee.FeeCap,
```

Comment:
I would recommend keeping these here and moving the estimateGas call up. Like so:

```go
var gasPrice, gasTipCap, gasFeeCap *big.Int

if ex.config.KeeperCheckUpkeepGasPriceFeatureEnabled() {
  gasPrice = foo
   ...
   ```